src/SignalR/clients/csharp/Client.SourceGenerator/src/HubProxyGenerator.Parser.cs

```diff
@@ -0,0 +1,167 @@
+                    {
+                        var methodSpec = new MethodSpec
+                        {
+                            Name = member.Name, FullyQualifiedReturnTypeName = member.ReturnType.ToString()
+                        };
+
+                        if (member.ReturnType is INamedTypeSymbol {Arity: 1} rtype)
+                        {
+                            methodSpec.InnerReturnTypeName = rtype.TypeArguments[0].ToString();
+                        }
+
+                        if (member.ReturnType is INamedTypeSymbol {Arity: 1, Name: "Task"} a
+                            && a.TypeArguments[0] is INamedTypeSymbol {Arity: 1, Name: "ChannelReader"} b)
+                        {
+                            methodSpec.Stream = StreamSpec.ServerToClient & ~StreamSpec.AsyncEnumerable;
+                            methodSpec.InnerReturnTypeName = b.TypeArguments[0].ToString();
+                        }
+                        else if (member.ReturnType is INamedTypeSymbol {Arity: 1, Name: "IAsyncEnumerable"} c)
+                        {
+                            methodSpec.Stream = StreamSpec.ServerToClient | StreamSpec.AsyncEnumerable;
+                            methodSpec.InnerReturnTypeName = c.TypeArguments[0].ToString();
+                        }
+                        else
+                        {
+                            methodSpec.Stream = StreamSpec.None;
+                        }
+
+                        // Generate spec for each argument
+                        foreach (var parameter in member.Parameters)
+                        {
+                            var argumentSpec = new ArgumentSpec
+                            {
+                                Name = parameter.Name, FullyQualifiedTypeName = parameter.Type.ToString()
+                            };
+
+                            methodSpec.Arguments.Add(argumentSpec);
+
+                            switch (parameter.Type)
+                            {
+                                case INamedTypeSymbol {Arity: 1, Name: "ChannelReader"}:
+                                    methodSpec.Stream |= StreamSpec.ClientToServer;
+                                    break;
+                                case INamedTypeSymbol {Arity: 1, Name: "IAsyncEnumerable"}:
+                                    methodSpec.Stream |= StreamSpec.ClientToServer;
+                                    break;
+                            }
+                        }
+
+                        // Validate return type
+                        if (!methodSpec.Stream.HasFlag(StreamSpec.ServerToClient))
+                        {
+                            if (methodSpec.Stream.HasFlag(StreamSpec.ClientToServer) &&
+                                member.ReturnType is not INamedTypeSymbol {Arity: 0, Name: "Task" or "ValueTask"})
+                            {
+                                _context.ReportDiagnostic(Diagnostic.Create(
```

Comment:
`var res = await HubConnection.InvokeAsync("Method", channel.Reader);`
This is supported, would look like:
```
interface I
{
    Task<int> Method(ChannelReader);
}
var res = await connection.Method(channel.Reader);
```

Reply:
Done. This should generate fine now.
```csharp
        public System.Threading.Tasks.Task<int> ClientToServerStreamButAlsoReturnValue(System.Threading.Channels.ChannelReader<float> reader)
        {
            return this.conn.InvokeAsync<int>("ClientToServerStreamButAlsoReturnValue", reader);
        }
```